pinot-server/src/main/java/org/apache/pinot/server/starter/grpc/PinotQueryService.java

```diff
@@ -18,6 +18,57 @@
  */
 package org.apache.pinot.server.starter.grpc;
 
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
 public class PinotQueryService {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PinotQueryService.class);
+
+  private final int port;
+  private final Server server;
+
+  // TODO: pass config, and parameters to initialize handler
+  public PinotQueryService(int port) {
+    this.port = port;
+    server = ServerBuilder.forPort(port)
+        .addService(new PinotQueryHandler())
+        .build();
+
+  }
+
+  /** Start serving requests. */
+  public void start() throws IOException {
+    server.start();
+    LOGGER.info("Server started. Listening on {}", port);
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```

Comment:
Not sure if this shutdown hook is necessary

Reply:
Removed, can always add it back if needed.